graphql/src/main/java/com/linecorp/armeria/server/graphql/GraphQLServiceBuilder.java

```diff
@@ -0,0 +1,218 @@
+        return instrumentation(ImmutableList.copyOf(requireNonNull(instrumentations, "instrumentations")));
+    }
+
+    /**
+     * Sets the {@link Instrumentation}.
+     */
+    public GraphQLServiceBuilder instrumentation(Iterable<Instrumentation> instrumentations) {
+        instrumentationBuilder.addAll(requireNonNull(instrumentations, "instrumentations"));
+        return this;
+    }
+
+    /**
+     * Sets the {@link GraphQL.Builder} consumer.
+     */
+    public GraphQLServiceBuilder configure(Consumer<GraphQL.Builder>... configurers) {
+        return configure(ImmutableList.copyOf(requireNonNull(configurers, "configurers")));
+    }
+
+    /**
+     * Sets the {@link GraphQL.Builder} consumer.
+     */
+    public GraphQLServiceBuilder configure(Iterable<Consumer<GraphQL.Builder>> configurers) {
+        configurerBuilder.addAll(requireNonNull(configurers, "configurers"));
+        return this;
+    }
+
+    /**
+     * Creates a {@link GraphQLService}.
+     */
+    public GraphQLService build() {
+        final TypeDefinitionRegistry registry = typeDefinitionRegistry(schemaFileBuilder.build());
+
+        final RuntimeWiring runtimeWiring = buildRuntimeWiring(runtimeWiringConfiguratorBuilder.build());
+        GraphQLSchema schema = new SchemaGenerator().makeExecutableSchema(registry, runtimeWiring);
+        final List<GraphQLTypeVisitor> typeVisitors = typeVisitorBuilder.build();
+        for (GraphQLTypeVisitor typeVisitor : typeVisitors) {
+            schema = SchemaTransformer.transformSchema(schema, typeVisitor);
+        }
+
+        GraphQL.Builder builder = GraphQL.newGraphQL(schema);
+        final List<Instrumentation> instrumentations = instrumentationBuilder.build();
+        if (!instrumentations.isEmpty()) {
+            builder = builder.instrumentation(new ChainedInstrumentation(instrumentations));
+        }
+
+        final List<Consumer<GraphQL.Builder>> configurers = configurerBuilder.build();
+        for (Consumer<GraphQL.Builder> configurer : configurers) {
+            configurer.accept(builder);
+        }
+        return new DefaultGraphQLService(builder.build(), dataLoaderRegistry);
+    }
+
+    private static TypeDefinitionRegistry typeDefinitionRegistry(List<File> schemaFiles) {
+        final TypeDefinitionRegistry registry = new TypeDefinitionRegistry();
+        final SchemaParser parser = new SchemaParser();
+        if (schemaFiles.isEmpty()) {
+            schemaFiles = defaultSchemaFiles();
+        }
+        if (schemaFiles.isEmpty()) {
+            throw new IllegalStateException("Not found schema file(s)");
+        }
+        schemaFiles.forEach(it -> registry.merge(parser.parse(it)));
+        return registry;
+    }
+
+    private static RuntimeWiring buildRuntimeWiring(
+            List<RuntimeWiringConfigurator> runtimeWiringConfigurators) {
+        final Builder runtimeWiringBuilder = RuntimeWiring.newRuntimeWiring();
```

Comment:
Don't we need to raise `IllegalStateException` if `runtimeWiringConfigurators` is empty?

Reply:
Since we provide `configure(Consumer<GraphQL.Builder>... configurers)`, I am not sure if we need to raise an exception.

Reply:
I see, then let's leave it as it is.